docs/DeveloperGuide.md

```diff
@@ -1,34 +1,630 @@
+design of **Finux** and describes the current implementation details of our main features
+with the thought process behind each of them. The current edition of this guide is meant for
+the `v2.1` release.
+
+### 1.2 Using this Guide
+
+Throughout the guide, you might encounter several icons, refer to the following consolidate
+list of icons:
+
+> 💡 It indicates that this is an important piece of information.\
+> ❗  It indicates that this is a warning and you should be aware of it while developing.\
+> 📝 It indicates that this is a tip and you may find it useful when using our application or guide.
+
+> 📝 Text that are blue like this [example(jumps to 1.2)](#12-using-this-guide), are clickable links
+> that will move you to the relevant section of the guide.
+
+## 2. Setting Up
+
+---
+
+### 2.1 Setting up the project in your computer
+
+Ensure that you have the following installed:
+* JDK 11
+* Intellij IDEA (Highly Recommended)
+
+Firstly, **fork** this repo and **clone** a copy into your computer.
+
+If you plan to use IntelliJ IDEA:
+1. **Ensure IntelliJ is configured to develop in JDK 11.**
+   1. If you are not at the welcome screen, click `File` > `Close Projects` to close any currently opened projects.
+   1. Click on `Configure` > `Structure for New Projects` > `Project Settings` > `Project`, ensure
+      the `Project SDK` is using **JDK 11**.
+1. **Import the project as a Gradle Project.**
+   1. Click on `Import Project` and locate the `build.gradle` file and select it. Click `OK`.
+   1. If prompted, choose to `Open as Project` (not `Open as File`).
+   1. Click `OK` to accept the default settings but do ensure that the selected version of `Gradle JVM`
+      matches the JDK being used for the project.
+   1. Wait for the importing process to finish (it could take a few minutes).
+1. **Verify the setup.**
+   1. After importing successfully, locate the `src/main/java/seedu.duke/Duke.java` file, right click it, 
+      and choose `Run...`. If the setup is correct, you should see the following:
+   ![Login_Page](img/LoginPageExampleOutput.png)
+
+### 2.2 Before writing code
+
+#### 2.2.1 Configuring the Coding Style
+
+If you are using IDEA, follow this guide 
+[IDEA: Configuring the code style](https://se-education.org/guides/tutorials/intellijCodeStyle.html)
+to setup IDEA's coding style to match ours.
+
+#### 2.2.2 Set up CI
+
+There is no set up required as the project comes with GitHub Actions config files, located in `.github/workflows`
+folder. When GitHub detects these files, it will run the CI for the project automatically at each push to the master
+branch or to any PR.
+
+#### 2.2.3 Learn the Design
+
+Before starting to write any code, we recommend that you have a look at Finux's overall design by reading
+[Finux's Architecture](#31-architecture) section.
+
+## 3. Design
+
+---
+
+### 3.1 Architecture
+
+![ArchitectureDiagram](img/ArchitectureDiagram.png)\
```

Comment:
Should the arrows coming out of the "Finux" box be connected to some of the other boxes?

Reply:
It is intended, Finux initializes all the other components.
Thanks for asking!